### PR TITLE
feat(error tracking): rust consumer skeleton

### PR DESCRIPTION
**Pull request by oliverb123**
## Problem

Errors. Track em, tag em, put 'em in a stew. This is step one to getting a basic consumer deployed that can rip events off the exception-ingest topics and do something with them.

@bretthoerner tagging you and @frankh because I've yanked out some common kafka and event-type stuff from capture and propdefs, and it'd be good to get your eyes on that specifically (you can ignore everything else).

### CODE REVIEW

**Comment by bretthoerner** [comment]
> because I've yanked out some common kafka and event-type stuff from capture and propdefs, and it'd be good to get your eyes on that specifically (you can ignore everything else)

I checked these. Just to verify, this was mostly code moving (plus I guess having a container that holds metadata + event in Capture) right? If so LGTM. (I don't want to approve or it'll take the request away from others IIRC.)

**Comment by oliverb123** [comment]
@bretthoerner yep, plus the change to manual offset storing in propdefs (but that shouldn't matter really). Appreciate you taking a look, will wait on @neilkakkar and @daibhin to take a look, then merge soonish so I can get a deployment PR up.

File: rust/error-tracking/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Error tracking
+
+You throw 'em, we catch 'em.
```
Comment (neilkakkar) [inline rust/error-tracking/README.md:3]: 😂

**Review by neilkakkar — APPROVED** [review]
seems reasonable to me, wait for reviews for capture/prop-defs though!

File: rust/common/types/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Common types
+
+For types used across our projects, like events, persons, etc. Each time you go to copy a type from somewhere, but it here instead.
```
Comment (daibhin) [inline rust/common/types/README.md:3]:
```suggestion
For types used across our projects, like events, persons, etc. Each time you go to copy a type from somewhere, put it here instead.
```

**Review by daibhin — APPROVED** [review]

**Comment by daibhin** [comment]
> I don't want to approve or it'll take the request away from others IIRC

@bretthoerner I believe that only applies if a team is tagged. Looks like the other individual tag-ees stay on the PR even if approved by one

**Pull request merged**